### ISSUE TITLE
Working on the new dataset of untrimmed videos with multi-labels

### ISSUE THREAD

**Issue by Powercoder64** [open-mmlab/mmaction2#1316]
Hello,

Thanks for your awesome codes. I am trying to use some of your methods on my own dataset. My dataset includes multiple untrimmed long videos, and each long video includes multiple consecutive action labels. Each video frame also includes multiple labels. 
Based on my type of dataset, which dataset format I should follow? Is there any code for this type of data preparation? I found a code but it seems there is no part for the label preparation part. Also, which types of methods work in my type of dataset.

Would you please help me in this case?

**Comment by GriesserP** [comment]
Hello ,
I have the same problem (i.e. long videos with multiple lables on them), what I have done to finetune recognizers is that I extracted "clips" of 5sec from my long videos using ffmpeg. The extracted clips are centered around action of interest. Thus I have a proper "Videodataset" to finetune the recognizerd. The problem is for inference. As I want a prediction for each second of video, what I do is extract 5sec clips around each second of my video, extract the frames and (without using the datapreparation pipeline from mmaction) pass the extracted frames directly as input to the finetunded model. But I don't think this a proper way to handle the inference, I am trying to get a better solution that would use their datapreparation pipeline.

Hope it helps, and if someone has any suggestion on how to handle datasets of long videos, it's welcome

**Comment by Powercoder64** [comment]
Thanks for your answer. in my case

**Comment by Powercoder64** [comment]
> Hello , I have the same problem (i.e. long videos with multiple lables on them), what I have done to finetune recognizers is that I extracted "clips" of 5sec from my long videos using ffmpeg. The extracted clips are centered around action of interest. Thus I have a proper "Videodataset" to finetune the recognizerd. The problem is for inference. As I want a prediction for each second of video, what I do is extract 5sec clips around each second of my video, extract the frames and (without using the datapreparation pipeline from mmaction) pass the extracted frames directly as input to the finetunded model. But I don't think this a proper way to handle the inference, I am trying to get a better solution that would use their datapreparation pipeline.
> 
> Hope it helps, and if someone has any suggestion on how to handle datasets of long videos, it's welcome

Thanks for your answer. In my case, I cannot cut the videos based on the actions of interest since each frame has multiple labels. If I cut, then, the size of data becomes so huge since there will be a lot of repeated frames (or maybe I need to share the frames across the labels, which needs a complex design). As you said, I feel that during training it is not possible to feed the whole long video since it becomes computationally expensive. Also, it seems the current codes do not support that. If I am wrong, please the developers correct me?

**Comment by GriesserP** [comment]
For the multiple label problem:
If you have multiple labels per frame/videos I don't think a recognizer model would suit your problem as their purpose is to do single classification of clips. Maybe a workaround woul be to put, in the training label ".txt" file, multiple entries for the same video, but each time with a different class. Then during inference, instead of looking at the highest output neuron you look at for example all neurons above a confidence of 0.8. I am not sure this is a proper way of handling your problem, but at least it would give you some results

**Comment by edybk** [comment]
> Hello , I have the same problem (i.e. long videos with multiple lables on them), what I have done to finetune recognizers is that I extracted "clips" of 5sec from my long videos using ffmpeg. The extracted clips are centered around action of interest. Thus I have a proper "Videodataset" to finetune the recognizerd. The problem is for inference. As I want a prediction for each second of video, what I do is extract 5sec clips around each second of my video, extract the frames and (without using the datapreparation pipeline from mmaction) pass the extracted frames directly as input to the finetunded model. But I don't think this a proper way to handle the inference, I am trying to get a better solution that would use their datapreparation pipeline.
> 
> Hope it helps, and if someone has any suggestion on how to handle datasets of long videos, it's welcome

I have done something similar to finetune the recognizers on my dataset. I'm currently looking at ways to use the fine-tuned recognizers for frame-level inference. Were you able to find a way to do it using the mmaction preparation pipeline?

**Comment by kennymckormick** [comment]
> > Hello , I have the same problem (i.e. long videos with multiple lables on them), what I have done to finetune recognizers is that I extracted "clips" of 5sec from my long videos using ffmpeg. The extracted clips are centered around action of interest. Thus I have a proper "Videodataset" to finetune the recognizerd. The problem is for inference. As I want a prediction for each second of video, what I do is extract 5sec clips around each second of my video, extract the frames and (without using the datapreparation pipeline from mmaction) pass the extracted frames directly as input to the finetunded model. But I don't think this a proper way to handle the inference, I am trying to get a better solution that would use their datapreparation pipeline.
> > Hope it helps, and if someone has any suggestion on how to handle datasets of long videos, it's welcome
> 
> I have done something similar to finetune the recognizers on my dataset. I'm currently looking at ways to use the fine-tuned recognizers for frame-level inference. Were you able to find a way to do it using the mmaction preparation pipeline?

I think you can check our long video demo, which is in the `demo` folder.
